runtime/tests/jvmtitests/src/com/ibm/jvmti/tests/fieldwatch/fw001.c

```diff
@@ -0,0 +1,233 @@
+ * Secondary Licenses when the conditions for such availability set
+ * forth in the Eclipse Public License, v. 2.0 are satisfied: GNU
+ * General Public License, version 2 with the GNU Classpath
+ * Exception [1] and GNU General Public License, version 2 with the
+ * OpenJDK Assembly Exception [2].
+ *
+ * [1] https://www.gnu.org/software/classpath/license.html
+ * [2] http://openjdk.java.net/legal/assembly-exception.html
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0 WITH Classpath-exception-2.0 OR LicenseRef-GPL-2.0 WITH Assembly-exception
+ *******************************************************************************/
+
+#include <errno.h>
+#include "jvmti_test.h"
+#include <stdio.h>
+#include <string.h>
+
+jvmtiEnv *globalEnv;
+jfieldID currentFieldBeingWatched;
+const char * testClassName;
+jstring testClassString;
+
+
+/* Find the correct API to increment the number of reports we have seen and return the new value. */
+jint
+incAndGetNumReports(JNIEnv* jni_env)
+{
+	jmethodID mID;
+	jclass clazz = (*jni_env)->FindClass(jni_env, testClassName);
+
+	if (NULL == clazz) {
+		printf("Unable to find class: %s\n", testClassName);
+		return JNI_ERR;
+	}
+
+	mID = (*jni_env)->GetStaticMethodID(jni_env, clazz, "incrementNumReports", "()I");
+	if (NULL == mID) {
+		printf("Unable to find methodID for class: %s and method: %s\n", testClassName, "incrementNumReports");
+		return JNI_ERR;
+	}
+
+	return (*jni_env)->CallStaticIntMethod(jni_env, clazz, mID);
+}
+
+/* Callback routine invoked by the VM to report a field read event. */
+void JNICALL
+fieldAccessCallback(jvmtiEnv *jvmti_env, JNIEnv* jni_env, jthread thread, jmethodID method, jlocation location, jclass field_klass, jobject object, jfieldID field)
+{
+	/* Only increment the count if the field ID matches with the same field we added the watch for. */
+	if (currentFieldBeingWatched == field) {
+		jint numReports = incAndGetNumReports(jni_env);
+		if (numReports >= JNI_ERR) {
+			printf("Field Access Report #%d: method: %p location %d. Class: %p, fieldID: %p\n", numReports, method, (jint)location, field_klass, field);
+		}
+		else {
+			printf("**Error**: Invalid value for numReports. Return Code: %d\n", numReports);
+		}
+	}
+}
+
+/* Callback routine invoked by the VM to report a field write event. */
+void JNICALL
+fieldModificationCallback(jvmtiEnv *jvmti_env, JNIEnv* jni_env, jthread thread, jmethodID method, jlocation location, jclass field_klass, jobject object, jfieldID field, char signature_type, jvalue new_value)
+{
+	/* Only increment the count if the field ID matches with the same field we added the watch for. */
+	if (currentFieldBeingWatched == field) {
+		jint numReports = incAndGetNumReports(jni_env);
+		if (numReports >= JNI_ERR) {
+			printf("Field Modification Report #%d: method: %p location: %d, fieldID: %p, signature: %c\n", numReports, method, (jint)location, field, signature_type);
+		}
+		else {
```

Comment:
Should be on the previous line.

Reply:
So just to confirm, it should be :
```
else
{

}
```

Right?

Reply:
I think what is meant is:

```
} else {
...
}
```

Reply:
@fjeremic is correct.